src/queue.js

```diff
@@ -25,17 +25,26 @@ export default class Queue {
         this.tail = node
       }
 
-      this.length += 1
-
+      this.size += 1
       return node
   }
 
   dequeue() {
-    
+    if (this.size === 0) return null
+    const node = this.head
+    this.head = node.next
+    node.next = null
+    this.head.previous = null
```

Comment:
A `Queue`'s `Node`s should not have a `previous` field (this is only used for doubly linked lists).